Ensure bids() is called with identifying entities

pvandyken opened the pull request:
Calls to bids with just datatype and/or prefix are indistinguishable from calls to bids with just suffix. Pybids would parse these paths as having suffixes. Thus, we prevent this types of calls altogether

Resolves #193



## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [x] Changes pass the unit tests
- [x] I have included necessary documentation or comments (as necessary)
- [x] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

tkkuehn commented on snakebids/core/construct_bids.py:
```diff
@@ -1,18 +1,19 @@
-    suffix: str = None,
-    subject: str = None,
-    session: str = None,
+    root: str | Path | None = None,
```
Nitpick: I prefer the `Optional[str | Path]` syntax.

kaitj commented on snakebids/utils/sb_itertools.py:
```diff
@@ -30,6 +30,15 @@ def unpack(iterable: Iterable[T], default: Sequence[T]) -> Iterable[T]:
 
 # pylint: disable=invalid-name
 def drop(n: int, iterable: Iterable[T]) -> Iterable[T]:
```
Should not prevent a merge, but it may be helpful to have this be slightly more descriptive. `drop` at a glance feels like it should be specific (e.g. `drop(7, range(10) -> [0, 1, 2, 3, 4, 5, 6, 8, 9, 10]`). Might be at the expense of a slightly longer name.

pvandyken replied:
Interesting, I chose the name _à la_ `more_itertools.take`, which has the opposite function. As the function takes an `int` but can act on any iterable, it doesn't give me any impression it acts as a value filter rather than an index filter.

I'm electing right now to leave this, but feel free to push back

kaitj replied:
Sounds good - I think I am more use to the `drop` in `pandas`. In any case, its not an issue. PR looks good to me otherwise.

pvandyken replied:
Ah, that makes sense

pvandyken commented:
Added another push updating the error message to be more detailed.

Also realized I made calls to bids with no args (`bids()`) illegal, but the docstring says such should return an empty string (`''`), so I fixed that
